src/apis/emscripten/env.rs

```diff
@@ -9,7 +9,18 @@ use std::mem;
 use std::os::raw::c_char;
 
 use super::utils::{allocate_on_stack, copy_cstr_into_wasm, copy_terminated_array_of_cstrs};
-use crate::webassembly::Instance;
+use crate::runtime::Instance;
+use crate::webassembly::instance::EmscriptenData;
+
+impl Instance {
+    pub fn memory_offset_addr(&self, index: usize, offset: usize) -> *const usize {
+        unimplemented!("TODO replace this stub")
+    }
+
+    pub fn emscripten_data(&self) -> &'static mut Option<EmscriptenData> {
```

Comment:
This still needs to be figured out. I don't think we should put this behind an `Option`, since we know that the emscripten data will be there or not. Something to think about.

Reply:
@lachlansneff Could the emscripten functions just be looked up at runtime? `exports.get("malloc")` ? Or maybe some hooks could be provided after a module/instance is created to do environment setup.